src/Simple.OData.Client.Core/Adapter/ResponseReaderBase.cs

```diff
@@ -131,6 +131,17 @@ protected void EndNavigationLink(Stack<ResponseNode> nodeStack)
                         d[FluentCommand.AnnotationsLiteral] = linkNode.Entry.Annotations;
                     }
                 }
+                else if (linkValue is IEnumerable<AnnotatedEntry> annotatedEntries)
+                {
+                    linkValue = annotatedEntries.Select(x =>
```

Comment:
This needs a `.ToArray()` otherwise there will be a strong reference to the items in the closure when they're added to the `AnnotatedEntry`'s `IDictionary<string, object>` below.

Reply:
You mean after the Select() statement?

Reply:
Yes, sorry annotating multi-line source isn't always obvious in GH.

Reply:
Thanks for the feedback, fixed.